src/main/java/io/fabric8/maven/docker/service/BuildService.java

```diff
@@ -60,14 +61,55 @@
      * @throws DockerAccessException
      * @throws MojoExecutionException
      */
-    public void buildImage(ImageConfiguration imageConfig, ImagePullManager imagePullManager, BuildContext buildContext)
+    public void buildImage(ImageConfiguration imageConfig, ImagePullManager imagePullManager, BuildContext buildContext, File buildArchiveFile)
             throws DockerAccessException, MojoExecutionException {
 
         if (imagePullManager != null) {
             autoPullBaseImage(imageConfig, imagePullManager, buildContext);
         }
 
-        buildImage(imageConfig, buildContext.getMojoParameters(), checkForNocache(imageConfig), addBuildArgs(buildContext));
+        buildImage(imageConfig, buildContext.getMojoParameters(), checkForNocache(imageConfig), addBuildArgs(buildContext), buildArchiveFile);
+    }
+
+    /**
+     * Create docker archive for building image
+     *
+     * @param imageConfiguration image configuration
+     * @param buildContext docker build context
+     * @param buildArchiveOnlyFlagOrBuildArchiveLocation build archive only flag, it can have values TRUE or FALSE and also
+     *                               it can hold path to archive where it might get copied over
+     * @return tarball for docker image
+     * @throws MojoExecutionException in case any exception comes during building tarball
+     */
+    public File buildArchive(ImageConfiguration imageConfiguration, BuildContext buildContext, String buildArchiveOnlyFlagOrBuildArchiveLocation)
+            throws MojoExecutionException {
+        String imageName = imageConfiguration.getName();
+        ImageName.validate(imageName);
+        BuildImageConfiguration buildConfig = imageConfiguration.getBuildConfiguration();
+        MojoParameters params = buildContext.getMojoParameters();
+
+        if (buildConfig.getDockerArchive() != null) {
+            return buildConfig.getAbsoluteDockerTarPath(params);
+        }
+        long time = System.currentTimeMillis();
+
+        File dockerArchive = archiveService.createArchive(imageName, buildConfig, params, log);
+        log.info("%s: Created %s in %s", imageConfiguration.getDescription(), dockerArchive.getName(), EnvUtil.formatDurationTill(time));
+
+        // Copy created tarball to directory if specified
+        if (buildArchiveOnlyFlagOrBuildArchiveLocation != null && !buildArchiveOnlyFlagOrBuildArchiveLocation.isEmpty()) {
```

Comment:
Can we extract that to a method ? Also, remember, when its empty its the same as false